### PR TITLE
#574: add instructions about how to build with Bazel

### LINKED ISSUE #574: Add Installation instructions for Bazel.
The install doc is empty which needs to be filled in with details.

https://github.com/open-telemetry/opentelemetry-cpp/blob/main/INSTALL.md

Similar issue for cmake: #564

### PR DESCRIPTION
**Pull request by iblancasa**
Fixes #574

## Changes
Add some instructions about how to build with Bazel.

### CODE REVIEW

**Comment by lalitb** [comment]
Thanks @iblancasa for the PR. We can get some feedback from @eyjohn and @jsuereth also here. As @eyjohn had some plans to reorganize the build/install document for cmake and bazel.

File: INSTALL.md
Comment (maxgolov) [inline INSTALL.md:172]: Technically the exclusion here isn't right. It all builds in its entirety right now, including `otlp` - which is going to be the main thing. Helper script for windows:
https://github.com/open-telemetry/opentelemetry-cpp/blob/414a7314f7ab343533042f70eaea5810764fdd5d/tools/build-bazel.cmd#L4

**Comment by maxgolov** [comment]
I think you should be good to re-add `otlp` exporter. If you fix that, it looks good then. Approved with suggestion.

File: INSTALL.md
Comment (maxgolov) [inline INSTALL.md:172]:
```suggestion
   bazel build //...
```

Comment (ThomsonTan) [reply INSTALL.md:172]: How about just invoke `tools/buildbazel.cmd` here which will keep both consistent?

**Review by maxgolov — APPROVED** [review]

**Comment by maxgolov** [comment]
> CI / markdown-lint (pull_request) Failing after 7s — markdown-lint

Please run the linter.

File: INSTALL.md
Comment (maxgolov) [reply INSTALL.md:172]: Maybe we need to add a script for Linux then, like `tools/build-bazel.sh` . But I actually like this document too.

File: INSTALL.md
Comment (maxgolov) [inline INSTALL.md:149]: Can we somehow omit `3.7.2.` here. It's 10 seconds to follow the guidelines, even if we don't mention the version explicitly. Maybe suggesting to check what version is being used:
https://github.com/open-telemetry/opentelemetry-cpp/blob/414a7314f7ab343533042f70eaea5810764fdd5d/.bazelversion#L1

Something like: **Please install the exact version of bazel mentioned in `.bazelversion`** ?


mgolovanov@xps:/mnt/c/work/opentelemetry-cpp.main$ bazel build //...
ERROR: The project you're trying to build requires Bazel 3.7.2 (specified in /mnt/c/work/opentelemetry-cpp.main/.bazelversion), but it wasn't found in /usr/bin.

You can install the required Bazel version via apt:
  sudo apt update && sudo apt install bazel-3.7.2

File: INSTALL.md
Comment (maxgolov) [reply INSTALL.md:172]: OTLP-gRPC takes forever to build though 😄

**Review by lalitb — APPROVED** [review]

**Review by eyjohn — APPROVED** [review]
Thanks for this contribution, it covers the process well.

I still have (slow) plans to restructure the install/usage docs focused on using otel-cpp in projects (which may/not include installing it) but I think this will build on top of these base instructions.

**Comment by lalitb** [comment]
@iblancasa - This is good to merge once the CI check for markdown lint is fixed. CI uses [markdownlint-cli](https://www.npmjs.com/package/markdownlint-cli) to validate the document using [this ](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/.markdownlint.json) configuration.

File: INSTALL.md
Comment (jsuereth) [inline INSTALL.md:152]: Isn't this more useful for contributors, less so installation?


E.g. I'd expect most bazel users to want  to do something like pulling in OTel CPP via `WORKSPACE` definition using `http_archive` or some other mechanism.

Not that this isn't useful, just not how I'd expect folks to consume.

**Review by jsuereth — COMMENTED** [review]
One meta question on instructions, otherwise, thanks much for taking a crack at this!

File: INSTALL.md
Comment (jsuereth) [reply INSTALL.md:148]: Yeah, I think we should likely encourage folks to install/use Bazelisk, as it can auto-correct versions on the fly and leads to a much smoother developer experience when you move back/forth between different bazel projects.

File: INSTALL.md
Comment (jsuereth) [reply INSTALL.md:152]: e.g. https://github.com/GoogleCloudPlatform/opentelemetry-operations-cpp/blob/main/WORKSPACE#L46

Or you can see:

https://github.com/googleapis/google-cloud-cpp/pull/1230/files for an example usage of OpenCensus C++

Comment (lalitb) [reply INSTALL.md:152]: This is a good point. We can have a separate section (say) "Incorporating Into An Existing Project", which includes the above-mentioned steps, similar to how we document for `CMake`.

**Comment by lalitb** [comment]
@iblancasa  - we would like to pull these changes in. But the markdown-lint check is failing for now. Once you fix that, we can merge the changes.

**Pull request merged**
